scripts/actions/check-job-progress.js

```diff
@@ -100,6 +100,88 @@ const getBatchStatus = async ({ batchUid, jobId }) => {
   }
 };
 
+/**
+ * @param {Object[]} slugStatuses
+ * @param {Boolean} slugStatuses[].ok - Boolean flag signaling if translation deserialized. If it did, we can treat this translation as having completed.
+ * @param {String} slugStatuses[].slug - Slug representing file path translated.
+ * @param {String} slugStatuses[].jobId - Job id translation is associated with.
+ */
+const aggregateStatuses = (slugStatuses) => {
+  let totalFailures = 0;
+  let totalSuccesses = 0;
+  const jobStatuses = {};
+
+  slugStatuses.map(({ ok, jobId }) => {
```

Comment:
Super nitpicky: If we're not returning anything, perhaps this could be a `forEach` or [`for..of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).

Reply:
If you really wanted to get fancy, this could all just be a reduce:


```js
const aggregateStatuses = (slugStatuses) =>
  slugStatuses.reduce((acc, status) => {
    const { ok, jobId } = status;
    const {  totalFailures, totalSuccess, jobStatuses } = acc;

    if (!(jobId in jobStatuses)) {
      jobStatuses[jobId] = { successes: 0, failures: 0 };
    }

    if (ok) {
      totalSuccesses += 1;
      jobStatuses[jobId].successes += 1;
    } else {
      totalFailures += 1;
      jobStatuses[jobId].failures += 1;
    }

    return {  totalFailures, totalSuccess, jobStatuses };
  }, { totalFailures: 0, totalSuccess: 0, jobStatuses: {});
```

Probably overkill tho.

Reply:
swapped to forEach. i did leave the implementation as is, but am now more convinced of using reduce 🙈